custom_components/localtuya/climate.py

```diff
@@ -0,0 +1,220 @@
+    PRECISION_HALVES,
+    PRECISION_TENTHS,
+)
+
+from .const import (
+    CONF_TARGET_TEMPERATURE_DP,
+    CONF_CURRENT_TEMPERATURE_DP,
+    CONF_MAX_TEMP_DP,
+    CONF_MIN_TEMP_DP,
+    CONF_FAN_MODE_DP,
+    CONF_HVAC_MODE_DP,
+    CONF_TEMPERATURE_STEP,
+    CONF_PRECISION,
+)
+from .common import LocalTuyaEntity, async_setup_entry
+
+_LOGGER = logging.getLogger(__name__)
+
+TEMPERATURE_CELSIUS = "celsius"
+TEMPERATURE_FAHRENHEIT = "fahrenheit"
+
+
+def flow_schema(dps):
+    """Return schema used in config flow."""
+    return {
+        vol.Optional(CONF_TARGET_TEMPERATURE_DP): vol.In(dps),
+        vol.Optional(CONF_CURRENT_TEMPERATURE_DP): vol.In(dps),
+        vol.Optional(CONF_TEMPERATURE_STEP): vol.In(
+            [PRECISION_WHOLE, PRECISION_HALVES, PRECISION_TENTHS]
+        ),
+        vol.Optional(CONF_HVAC_MODE_DP): vol.In(dps),
+        vol.Optional(CONF_FAN_MODE_DP): vol.In(dps),
+        vol.Optional(CONF_MAX_TEMP_DP): vol.In(dps),
+        vol.Optional(CONF_MIN_TEMP_DP): vol.In(dps),
+        vol.Optional(CONF_PRECISION): vol.In(
+            [PRECISION_WHOLE, PRECISION_HALVES, PRECISION_TENTHS]
+        ),
+        vol.Optional(CONF_TEMPERATURE_UNIT): vol.In(
+            [TEMPERATURE_CELSIUS, TEMPERATURE_FAHRENHEIT]
+        ),
+    }
+
+
+class LocaltuyaClimate(LocalTuyaEntity, ClimateEntity):
+    """Tuya climate device."""
+
+    def __init__(
+        self,
+        device,
+        config_entry,
+        switchid,
+        **kwargs,
+    ):
+        """Initialize a new LocaltuyaClimate."""
+        super().__init__(device, config_entry, switchid, **kwargs)
+        self._state = None
+        self._target_temperature = None
+        self._current_temperature = None
+        self._hvac_mode = None
+        self._preset_mode = None
+        self._precision = PRECISION_TENTHS
+        if self.has_config(CONF_PRECISION):
+            self._precision = self._config[CONF_PRECISION]
+        print("Initialized climate [{}]".format(self.name))
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features = 0
+        if self.has_config(CONF_TARGET_TEMPERATURE_DP):
+            supported_features = supported_features | SUPPORT_TARGET_TEMPERATURE
+        if self.has_config(CONF_MAX_TEMP_DP):
+            supported_features = supported_features | SUPPORT_TARGET_TEMPERATURE_RANGE
+        if self.has_config(CONF_FAN_MODE_DP):
+            supported_features = supported_features | SUPPORT_FAN_MODE
+        return supported_features
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return self._precision
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        if self.has_config(CONF_TEMPERATURE_UNIT):
+            if self._config[CONF_TEMPERATURE_UNIT] == TEMPERATURE_FAHRENHEIT:
+                return TEMP_FAHRENHEIT
+        return TEMP_CELSIUS
```

Comment:
This will return `TEMP_CELSIUS` even if `CONF_TEMPERATURE_UNIT` is not specified, is that intentional? There are other, similar cases as well.

Reply:
Yeah, it was my way to provide a default value (Celsius) in case the Optional value is unset.

Reply:
The it's fine 👍